specs/0006-label-taxonomy.md

```diff
@@ -0,0 +1,137 @@
+# A Taxonomy for Issue and PR Labels
+
+This is a proposal for formalizing the labels we use in our
+repositories in order to make them more meaningful and more useful.
+
+## Problem Description
+
+We use issue and pull request labels to indicate certain metadata
+about these items. Some labels are used to position an issue along a
+scale (is this high or low priority? Is this a big or small issue?).
+Other labels are used to group together issues that share some common
+aspect ("These are website issues" or "These issues need better
+descriptions").
+
+Our collection of labels has grown organically over time, and as we
+have created new labels they have not always been propagated to all
+of our repositories. This leads to two problems:
+
+- We can't always use labels effectively in our sprint planning
+  because sometimes the labels aren't available when we want them.
+
+- It can be difficult to differentiate between different types of
+  labels.
+
+## Policy
+
+I would like to propose a common label taxonomy to adopt across all of
+our repositories. The labels will be organized around several axes,
+indicated by prefixes on the label name. For example, labels that
+indicate priority will live in the `priority` access, and will be
+named like `priority/critical` or `priority/medium`. We should start
+with the following axes that cover most of our existing labels:
+
+- **Type**
+
+  The `type` axis describes in general terms the sort of work required
+  to resolve the issue.
+
+  - `type/bug`
+  - `type/enhancement`
+  - `type/documentation`
+  - `type/research`
+  - `type/discussion`
+
+- **Status**
+
+  The `status` axis describes the current condition of the issue or
+  pull request. This is a multi-valued axis (e.g., an issue may be
+  both `accepted` and `blocked`).
+
+  - `status/blocked`
+  - `status/wontfix`
+  - `status/invalid`
+  - `status/accepted`
+
+- **Topic**
+
+  The `topic` axis is groups to group issues and pull requests that
+  share some common aspect.
+
+  - `topic/monitoring`
+  - `topic/gitops`
+  - `topic/policy`
+  - `topic/mghpcc`
+  - `topic/website`
+
+- **Flag**
+
+  The `flag` axis is used to draw attention to an issue or pull
+  request.
+
+  - `flag/help wanted`
+  - `flag/needs description`
+  - `flag/added post planning`
+  - `flag/good first issue`
+  - `flag/question`
+
+- **Priority**
+
+  The `priority` axis is used to describe the criticality of an issue
+  or pull request.
+
+  - `priority/medium`
+  - `priority/high`
+  - `priority/critical`
```

Comment:
When speaking, we frequently label things as low priority. And I am sure while searching for labels I'd go looking for a label that says "low". Is there a reason you omitted a "low priority" label?

Reply:
Mostly I felt that "no priority" was the same as "low priority". It seemed like unnecessary clutter. Do we label everything with a priority tag? If something is unlabeled, what does that mean...is it lower than "low priority"? Higher?

Reply:
that makes sense to me.

Reply:
A thing I quite liked about https://seantrane.com/posts/logical-colorful-github-labels-18230/ was the user of now, soon, 2day, with the implication that anything else was just regular or less.  My assumption being that 2day is likely breaking or added late to sprint.  This addresses the frequent ‘it was high priority last year’.

Reply:
> This addresses the frequent ‘it was high priority last year’.

I mean, does it? We can still have something tagged `now` or `2day` that's been hanging around for multiple sprints because everybody was involved in other things. That seems like the same issue, just with different names.

That said, I'm not tied to `medium/high/critical` either, so if you think the terminology from that post is more descriptive, I'm happy to change things.

Reply:
On the surface this seems like a good idea.  I worry about the UI as I don't want to have to refer to several documents in order to determine what the taxonomy du jour is.  A UI would be also useful for those of us who are dyslexia as we will probably misspell random taxonomy labels.

Reply:
PS: I actually prefer low priority over no priority - I much prefer to be explicit as opposed to being implicit.

Reply:
>  UI would be also useful for those of us who are dyslexia as we will probably misspell random taxonomy labels.

You never have to type in label names: you select them from a menu.